### ISSUE TITLE
Edit Post: More Options menu toggle buttons cause focus loss when toggled

### ISSUE THREAD

**Issue by aduth** [WordPress/gutenberg#14849]
Previously: https://github.com/WordPress/gutenberg/pull/14456#issuecomment-473403200

**Describe the bug**

When activating the toggleable menu items from the post editor's More Options menu, a focus loss occurs. Proceeding to press Tab will restore focus to the expected position (in Chrome at least); however, the focus loss should not occur.

**To reproduce**

1. Navigate to Posts > Add New
2. Click or otherwise activate the "More Options" menu in the top-right
3. Press <kbd>Spacebar</kbd> (assumes first menu item focused)
4. Note focus loss

**Expected behavior**

Focus should remain on the activated toggle button ("Top Toolbar"), indicated by the focus outline on the button.

**Desktop:**

 - OS: macOS Mojave 10.14.4
 - Browser: Chrome Version 73.0.3683.86 (Official Build) (64-bit)

**Additional context**

As noted at https://github.com/WordPress/gutenberg/pull/14456#issuecomment-473403200, this has to do with how the menu item elements become unmounted and remounted.

**Comment by noisysocks** [comment]
@gziolo: Are you still working on this bug? It's causing an E2E test failure in https://github.com/WordPress/gutenberg/pull/14851 so I'm planning on including a fix in https://github.com/WordPress/gutenberg/pull/14851.

**Comment by gziolo** [comment]
> Are you still working on this bug? It's causing an E2E test failure in #14851 so I'm planning on including a fix in #14851.

I planned to work on it, but I got completely distracted. Let me know when you have the fix ready. I will help to land it.

**Comment by gziolo** [comment]
Awesome work @noisysocks 🚀
